### ISSUE TITLE
Call mediator with an address endpoint fails when used within an api resource

### ISSUE THREAD

**Issue by GDLMadushanka** [wso2/micro-integrator#1073]
**Description:**

$Title. Simply try the https://docs.wso2.com/display/EI650/Developing+Cloud+Native+Integration sample and you'll get the below error [1]. When we used 'send' mediator instead of 'call' mediator, it works.

[1]

`[2019-08-02 02:53:52,531] [micro-integrator] ERROR - ClientUtils The system cannot infer the transport information from the /forward URL.
[2019-08-02 02:53:52,533] [micro-integrator] ERROR - Axis2Sender Unexpected error during sending message out
org.apache.axis2.AxisFault: The system cannot infer the transport information from the /forward URL.
	at org.apache.axis2.description.ClientUtils.inferOutTransport(ClientUtils.java:86)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.executeImpl(DynamicAxisOperation.java:116)
	at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
	at org.apache.synapse.core.axis2.Axis2FlexibleMEPClient.send(Axis2FlexibleMEPClient.java:634)
	at org.apache.synapse.core.axis2.Axis2Sender.sendOn(Axis2Sender.java:85)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.send(Axis2SynapseEnvironment.java:569)
	at org.apache.synapse.mediators.builtin.CallMediator.handleNonBlockingCall(CallMediator.java:275)
	at org.apache.synapse.mediators.builtin.CallMediator.mediate(CallMediator.java:122)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.rest.Resource.process(Resource.java:363)
	at org.apache.synapse.rest.API.process(API.java:431)
	at org.apache.synapse.rest.RESTRequestHandler.apiProcess(RESTRequestHandler.java:135)
	at org.apache.synapse.rest.RESTRequestHandler.dispatchToAPI(RESTRequestHandler.java:113)
	at org.apache.synapse.rest.RESTRequestHandler.process(RESTRequestHandler.java:71)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:325)
	at org.apache.synapse.core.axis2.SynapseMessageReceiver.receive(SynapseMessageReceiver.java:92)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ServerWorker.processNonEntityEnclosingRESTHandler(ServerWorker.java:338)
	at org.apache.synapse.transport.passthru.ServerWorker.processEntityEnclosingRequest(ServerWorker.java:383)
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:152)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)`

**Suggested Labels:**

Micro Integrator, Call mediator, Bug, Documentation

**Suggested Assignees:**

**Affected Product Version:**
Micro Integrator 650

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**
Try this and you'll get the error;
https://docs.wso2.com/display/EI650/Developing+Cloud+Native+Integration

**Related Issues:**

**Comment by Nirothipan** [comment]
@GDLMadushanka 
I tired with following configs
```xml
<?xml version="1.0" encoding="UTF-8"?>
<api xmlns="http://ws.apache.org/ns/synapse" name="customer" context="/original">
    <resource methods="POST">
        <inSequence>
            <log level="full"/>
            <property name="FORCE_POST_PUT_NOBODY" value="true" scope="axis2" type="BOOLEAN"/>
            <call>
                <endpoint>
                    <address uri="http://www.mocky.io/v2/5ed7d60932000019d9274cb8"/>
                </endpoint>
            </call>
            <respond/>
        </inSequence>
        <outSequence/>
        <faultSequence/>
    </resource>
</api>
```
In following products
- EI 6.5.0
- MI 1.1.0
- MI 1.2.0_alpha
- MI 1.1.0_beta3 ( the one in studio )
still the issue is not reproducible, can you tell what is the missing step here.

**Comment by Nirothipan** [comment]
Tested the issue with the artifacts from the sample. Still not reproducible.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<api context="/forward" name="forwardOrderApi" xmlns="http://ws.apache.org/ns/synapse">
    <resource methods="POST">
        <inSequence>
            <call>
                <endpoint>
                    <address uri="http://localhost:8290/order"/>
                </endpoint>
            </call>
            <respond/>
        </inSequence>
        <outSequence/>
        <faultSequence/>
    </resource>
</api>
```
```xml
<?xml version="1.0" encoding="UTF-8"?>
<api context="/order" name="OrderProcessApi" xmlns="http://ws.apache.org/ns/synapse">
    <resource methods="POST">
        <inSequence>
            <payloadFactory description="order placement payload" media-type="json">
                <format>{
	"orderDetails":$1,
	"orderID":"1a23456",
	"price":25.65,
	"status":"successful"
}</format>
                <args>
                    <arg evaluator="json" expression="$"/>
                </args>
            </payloadFactory>
            <log description="order result log" level="full"/>
            <respond description="respond with order placement details"/>
        </inSequence>
        <outSequence/>
        <faultSequence/>
    </resource>
</api>
```
Please re-open if you can provide more exact steps to reproduce this issue.

**Comment by gparedesprieto** [comment]
I'm using Api Manager 4.0.0, When I published an API from SOAP 1.1 text/xml.

By default the resource Mediation IN:
<property description="messageProperty" name="messageType" scope="axis2" type="STRING" value="application/soap+xml"/>

but I think is necesario change content type from application/soap+xml to text/xml ???

Please your help.
